pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -0,0 +1,647 @@
+    Hardie & Fader [2]_ [3]_ [4]_.
+
+    The Pareto/NBD model assumes churn times for the customer population (i.e., amount of time a customer is active)
+    follows a Gamma distribution,
+    and time between purchases is also Gamma-distributed while the customer is still active.
+
+    This model requires customer identifiers and that data be aggregated in RFM format via `clv.rfm_summary()` or
+    equivalent. Specifically, data must be summarized by recency, frequency, and T for each customer:
+
+    **recency**: Number of time periods between the customer's first and most recent purchases.
+    **frequency**: Number of repeat purchases per customer.
+
+    **T**: Number of time periods since the customer's first purchase. Model assumptions require T >= recency.
+
+    Please note this model is still experimental. See code examples in documentation if encountering fitting issues.
+
+    Parameters
+    ----------
+    data: pd.DataFrame
+        DataFrame containing the following columns:
+            * `frequency`: number of repeat purchases
+            * `recency`: time between the first and the last purchase
+            * `T`: time between the first purchase and the end of the observation period; model assumptions require T >= recency
+            * `customer_id`: unique customer identifier
+    model_config: dict, optional
+        Dictionary containing model parameters:
+            * `r`: Shape parameter of time between purchases for customer population; defaults to `pymc.Weibull.dist(alpha=2, beta=1)`
+            * `alpha`: Scale parameter of time between purchases for customer population; defaults to `pymc.Weibull.dist(alpha=2, beta=10)`
+            * `s`: Shape parameter of time until churn for customer population; defaults to `pymc.Weibull.dist(alpha=2, beta=1)`
+            * `beta`: Scale parameter of time until churn for customer population; defaults to `pymc.Weibull.dist(alpha=2, beta=10)`
+        If not provided, the model will use default priors specified in the `default_model_config` class attribute.
+    sampler_config: dict, optional
+        Dictionary of sampler parameters. Defaults to None.
+
+    Examples
+    --------
+        .. code-block:: python
+
+            import pymc as pm
+            from pymc_marketing.clv import ParetoNBDModel, rfm_summary
+
+            rfm_df = rfm_summary(raw_data,'id_col_name','date_col_name')
+
+            model = ParetoNBDModel(
+                data=rfm_df,
+                model_config={
+                r=pm.Weibull.dist(alpha=2,beta=1),
+                alpha=pm.Weibull.dist(alpha=2,beta=10),
+                s=pm.Weibull.dist(alpha=2,beta=1),
+                beta=pm.Weibull.dist(alpha=2,beta=10),
+                },
+                sampler_config={
+                    "draws": 1000,
+                    "tune": 1000,
+                    "chains": 2,
+                    "cores": 2,
+                    "nuts_kwargs": {"target_accept": 0.95},
+                },
```

Comment:
I don't think this is needed in the example. Less can be more!

Reply:
Sampler configurations can make or break the `ParetoNBDModel` as we've discovered in the dev notebook. However, the slow tests converge fine with the default configs, so I'll remove this for now and add recommendations in a future PR if issues arise.